skyportal/facility_apis/ztf.py

```diff
@@ -882,7 +881,64 @@ def custom_json_schema(instrument, user):
         return form_json_schema
 
 
-def fetch_observations(instrument_id, client, request_str):
+def fetch_depot_observations(instrument_id, session, depot_url, jd_start, jd_end):
+    """Fetch executed observations from a TAP client.
+    instrument_id : int
+        ID of the instrument
+    session : request.Session()
+        An authenticated request session.
+    depot_url : str
+        URL of the depot server
+    jd_start : float
+        JD of the start time of observations
+    jd_end : float
+        JD of the end time of observations
+    """
+
+    dfs = []
+
+    jds = np.arange(np.floor(jd_start), np.ceil(jd_end))
+    for jd in jds:
+        date = Time(jd, format='jd').datetime.strftime("%Y%m%d")
+        url = f'{depot_url}/{date}/ztf_recentproc_{date}.json'
+        r = session.head(url)
+
+        # file exists
+        if r.status_code == 200:
+            r = session.get(url)
+            obstable = pd.DataFrame(r.json())
+
+            if obstable.empty:
+                log(f'No observations for instrument ID {instrument_id} for JD: {jd}')
```

Comment:
This could produce quite a bit of logging outputs... are you sure we need this?

Reply:
Should be very rare to trigger I think.

Reply:
OK then